File: abc325/c/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i, n) for (int i = 0; i < n; i++)
#define REP(i, x, n) for (int i = x; i < n; i++)
using vi = vector<int>;
using vvi = vector<vi>;
using ll = long long;
using vl = vector<ll>;
using vll= vector<vl>;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
using vb = vector<bool>;
using si = set<int>;
using sl = set<ll>;

void dfs(int x, int y, vector<vector<bool>> &visited, vector<string> &board, int h, int w) {
    if (visited[x][y]) return;
    
    visited[x][y] = true;
    
    int dx[] = {0, 0, 1, -1, 1, 1, -1, -1};
    int dy[] = {1, -1, 0, 0, 1, -1, 1, -1};

    rep(i, 8) {
        int nx = x + dx[i], ny = y + dy[i];

        if (0 <= nx && nx < h && 0 <= ny && ny < w && board[nx][ny] == '#' && !visited[nx][ny]) {
            dfs(nx, ny, visited, board, h, w);
        }
    }
}

int count_sensor(vector<string> &board, int h, int w) {
    vector<vector<bool>> visited(h, vector<bool>(w, false));
    int count = 0;

    rep(x, h) {
        rep(y, w) {
            if (board[x][y] == '#' && !visited[x][y]) {
                count++;
                dfs(x, y, visited, board, h, w);
            }
        }
    }

    return count;
}

int main()
{
	int h, w;
    cin >> h >> w;
    vector<string> board(h);
    rep(i, h) {
        cin >> board[i];
    }

    cout << count_sensor(board, h, w) << endl;

    return 0;
}

File: abc325/c/main.py
import sys
sys.setrecursionlimit(10**6)

def dfs(x, y, visited, board, h, w):
    if visited[x][y]:
        return
    
    visited[x][y] = True
    
    dx = [0, 0, 1, -1, 1, 1, -1, -1]
    dy = [1, -1, 0, 0, 1, -1, 1, -1]

    for i in range(8):
        nx, ny = x + dx[i], y + dy[i]

        if 0 <= nx < h and 0 <= ny < w and board[nx][ny] == '#' and not visited[nx][ny]:
            dfs(nx, ny, visited, board, h, w)

def count_sensor(board, h, w):
    visited = [[False] * w for _ in range(h)]
    count = 0

    for x in range(h):
        for y in range(w):
            if board[x][y] == '#' and not visited[x][y]:
                count += 1
                dfs(x, y, visited, board, h, w)

    return count

h, w = map(int, input().split())
board = [list(input()) for _ in range(h)]

print(count_sensor(board, h, w))
